src/tir/ir/script/script_complete.cc

```diff
@@ -76,8 +76,15 @@ class ScriptCompleter : public StmtMutator {
     for (const auto& alloc_buffer : op->alloc_buffers) {
       buffer_var_map_->erase(alloc_buffer->data);
     }
+    // Get access detection mask
+    // 0 for provided region, 1 and 3 for need detect read, 2 and 3 for need detect write
+    int mask = 0;
+    auto it = op->annotations.find(attr::script_parsing_detect_access);
+    if (it != op->annotations.end()) {
+      mask = Downcast<IntImm>((*it).second)->value;
+    }
     // ignore root block or blocks which already has reads/writes regions
-    if (block->reads.empty() || block->writes.empty()) {
+    if (mask != 0) {
```

Comment:
Is there a reason we moved away from high level checks like this? the masking logic is way less clear from my PoV.

Reply:
The reason being the that the reads/write empty may not be a good indicator(since user might indicated the reads write being empty explicitly) and we need requests from the parser side.

Reply:
Can we potentially wrap the masking logic? I think its a bit of a leaky abstraction for end users to need to understand low level bit masking, happy to have this be a follow up item.

Reply:
This is an abstraction being used by the parser(during parsing), but not being used as part of standard TIR, so it might be fine here. Happy to change to two boolean flags if needed